### DISCUSSION TITLE
DummyThreads Appearing After Measurement Stops

### DISCUSSION THREAD

**Discussion by Did-Mas** [pymeasure/pymeasure#1192] [Q&A]
I'm relatively new to Python threading, and while working on a project, I noticed something unusual during debugging. After queuing a measurement, two threads appear (I assume one is the worker and the other the monitor). These threads disappear after finishing, but when the process completes, a DummyThread is also added. If many measurements are performed, these threads accumulate over time. What is their function?

**Comment by CasperSchippers** [comment]
That is actually a very good question to which I also don't have an actual answer. It _seems_ that these dummy threads are a consequence of using QThreads, which creates threads outside of the standard threading library; an answer on stackoverflow elaborates on this: https://stackoverflow.com/a/55778564.
